src/v/kafka/server/handlers/fetch.cc

```diff
@@ -390,6 +390,21 @@ static void fill_fetch_responses(
   std::vector<op_context::response_placeholder_ptr> responses,
   std::vector<std::unique_ptr<hdr_hist::measurement>> metrics) {
     auto range = boost::irange<size_t>(0, results.size());
+    if (unlikely(
+          results.size() != responses.size()
+          || results.size() != metrics.size())) {
+        // soft assert & recovery attempt
+        vlog(
+          klog.warn,
```

Comment:
should we log at error level? then we'll be sure to get a ducktape bad-log-lines trigger. and besides, this is supposed to be representing a violation of an invariant.

Reply:
oh I thought for a moment that warns are also checked by the bad-log-lines but they are not. Changing now